src/Compatibility/Microsoft.DotNet.ApiCompatibility/ErrorSuppression/Suppression.cs

```diff
@@ -54,6 +55,14 @@ static bool AreEqual(string? first, string? second)
                 => string.IsNullOrEmpty(first?.Trim()) && string.IsNullOrEmpty(second?.Trim()) || StringComparer.InvariantCultureIgnoreCase.Equals(first?.Trim(), second?.Trim());
         }
 
-        public override int GetHashCode() => HashCode.Combine(DiagnosticId?.ToLowerInvariant(), Target?.ToLowerInvariant(), Left?.ToLowerInvariant(), Right?.ToLowerInvariant(), IsBaselineSuppression);
+        public override int GetHashCode()
+        {
+            int hashCode = 1447485498;
+            hashCode = hashCode * -1521134295 + EqualityComparer<string>.Default.GetHashCode(DiagnosticId ?? string.Empty);
```

Comment:
This is changing the source value, it needs to use ToLowerInvariant as the suppressions are not case sensitive?